spec/registry.md

```diff
@@ -384,6 +384,158 @@ its _resolved value_ contains the implementation-defined integer value
+The options `unit` and `perUnit` MAY be used to override the units of an implementation-defined type,
+provided the units are compatible and the implementation supports conversion.
+
+The value of the _operand_'s `unit` SHOULD be either a string containing a
+valid [Unit Identifier](https://www.unicode.org/reports/tr35/tr35-general.html#unit-identifiers)
+or an implementation-defined unit type.
+
+A [Number Operand](#number-operands) without a `unit` _option_ results in a _Bad Operand_ error.
+
+> [!NOTE]
+> For example, in ICU4J, the type `com.ibm.icu.util.Measure` can be used
+> to set the `value` and `unit` (and optionally the `perUnit`).
+
+> [!NOTE]
+> For runtime environments that do not provide a ready-made data structure,
+> class, or type for unit values, the implementation ought to provide
+> a data structure, convenience function, or documentation on how to encode
+> the value, unit, and optionally per-unit for formatting.
+> For example, such an implementation might define a "unit operand"
+> to include a key-value structure with specific keys to be the
+> local unit operand, which might look like the following:
+> ```
+> {
+>    "value": 123.45,
+>    "unit": "kilometer",
+>    "perUnit": "hour"
+> }
+> ```
+
+#### Options
+
+Some options do not have default values defined in this specification.
+The defaults for these options are implementation-dependent.
+In general, the default values for such options depend on the locale, 
+the unit,
+the value of other options, or all of these.
+
+The following options and their values are required to be available on the function `:unit`:
+- `select`
+   -  `plural` (default)
+   -  `ordinal`
+   -  `exact`
+- `unit`
+   - valid [Unit Identifier](https://www.unicode.org/reports/tr35/tr35-general.html#unit-identifiers)
+     (no default)
+- `perUnit`
+  - valid [Unit Identifier](https://www.unicode.org/reports/tr35/tr35-general.html#unit-identifiers)
+    (no default)
+- `unitDisplay`
+  - `short` (default)
+  - `narrow`
+  - `long`
+- `compactDisplay` (this option only has meaning when combined with the option `notation=compact`)
+   - `short` (default)
+   - `long`
+- `notation`
+   - `standard` (default)
+   - `compact`
+- `numberingSystem`
+   - valid [Unicode Number System Identifier](https://cldr-smoke.unicode.org/spec/main/ldml/tr35.html#UnicodeNumberSystemIdentifier)
+     (default is locale-specific)
+- `signDisplay`
+   -  `auto` (default)
+   -  `always`
+   -  `exceptZero`
+   -  `negative`
+   -  `never`
+- `useGrouping`
+  - `auto` (default)
+  - `always`
+  - `never`
+  - `min2`
+- `minimumIntegerDigits`
+  - ([digit size option](#digit-size-options), default: `1`)
+- `minimumFractionDigits`
+  - ([digit size option](#digit-size-options))
+- `maximumFractionDigits`
+  - ([digit size option](#digit-size-options))
+- `minimumSignificantDigits`
+  - ([digit size option](#digit-size-options))
+- `maximumSignificantDigits`
+  - ([digit size option](#digit-size-options))
+
```

Comment:
This also needs the **usage** setting. See https://unicode-org.github.io/icu-docs/apidoc/released/icu4j/com/ibm/icu/number/NumberFormatterSettings.html

That is what is used to signal that the unit **should be** converted to the local units if possible. Without it, the unit should **not** be converted.

Reply:
```suggestion
- `usage`
```

Reply:
Where are the valid values defined? Should I define a grammar for these (appears to be hyphen separated ASCII tokens??)?

I haven't worked with units in NumberFormatter. My experience is limited to MeasureFormat. I have a sense that I should push this to post-46.1 and we start with design (preferably from someone on the implementation side) rather than forcing it here.

Reply:
The grammar is defined in https://www.unicode.org/reports/tr35/tr35-general.html#unit-identifiers, with the EBNF at https://www.unicode.org/reports/tr35/tr35-general.html#syntax

Reply:
Those are unit identifiers. Where is the definition for **_usage_** identifiers?

Reply:
Sorry, the valid usage identifiers are `category` attribute values of the element `unitPreferences`, which are in the file [`supplemental/units.xml`](https://github.com/unicode-org/cldr/blob/main/common/supplemental/units.xml).

For example, `blood-glucose` because it is in:
```
<unitPreferences category="concentration" usage="blood-glucose">
  <unitPreference regions="AG AI AO AU BA BG BH BM BN BW BY CA CH CM CN CZ DK DM EE FI FJ GB GD HK HR HU IE IM IS KE KN KW KZ LC LI LT LU LV ME MG MK MO MS MT MU MY MZ NA NL NO NZ OM PG RS RU SE SG SI SK TC TO UA UG VC VG VN VU ZA">millimole-per-liter</unitPreference>
  <unitPreference regions="001">milligram-ofglucose-per-deciliter</unitPreference>
</unitPreferences>
```

Note: there is a ticket for creating a regular validity/ data file for the usage values. It would make it cleaner to reference that, so I'll look at accelerating that into v46.1 so that MF2.0 in 46.1 can use a more direct reference.

Reply:
```suggestion
- `usage`
    - Well-formed and valid usage identifiers are defined in [Unicode Preferences](https://www.unicode.org/reports/tr35/tr35-info.html#unit-preferences).
```

I have a PR to make the target of the link clearer, in https://github.com/unicode-org/cldr/pull/4179. I don't foresee any problem merging that into 46.1.